enterprise/users/src/main/java/io/crate/auth/user/StatementPrivilegeValidator.java

```diff
@@ -84,30 +85,34 @@
 
 class StatementPrivilegeValidator implements StatementAuthorizedValidator {
 
-    private static final StatementVisitor VISITOR = new StatementVisitor();
-
+    private final UserLookup userLookup;
     private final User user;
 
-    StatementPrivilegeValidator(User user) {
+    StatementPrivilegeValidator(UserLookup userLookup, User user) {
```

Comment:
was there a specific reason to pass in the userLookup?

Reply:
It's to lookup the owner of the view, not the current user. 

Otherwise the `AnalyzedView` would have to contain a `User` owner object, which would require either Schemas or the RelationAnalyzer to depend on the UserLookup.